app/models/agents/ftpsite_agent.rb

```diff
@@ -92,6 +133,18 @@ def check
       end
     end
 
+    def receive(incoming_events)
+      return if interpolated['mode'] != 'write'
+      incoming_events.each do |event|
+        mo = interpolated(event)
+        string = Utils.value_at(event.payload, mo['data_path'])
+        string.encode!(interpolated['force_encoding'], invalid: :replace, undef: :replace) if interpolated['force_encoding'].present?
+        open_ftp(base_uri) do |ftp|
+          ftp.storbinary("STOR #{mo['filename']}", StringIO.new(string), Net::FTP::DEFAULT_BLOCKSIZE)
```

Comment:
Good point, i tried `"STOR #{mo['filename']}\r\nDELETE #{mo['filename']}"` in the code and it did not delete the file, so I think we are good.